salt/utils/win_network.py

```diff
@@ -0,0 +1,337 @@
+    ``win_interfaces`` function in ``salt.utils.network`` unchanged.
+    Returns:
+        dict: A dictionary of information about all interfaces on the system
+    '''
+    with salt.utils.winapi.Com():
+        c = wmi.WMI()
+        ifaces = {}
+        for iface in c.Win32_NetworkAdapterConfiguration(IPEnabled=1):
+            ifaces[iface.Description] = dict()
+            if iface.MACAddress:
+                ifaces[iface.Description]['hwaddr'] = iface.MACAddress
+            if iface.IPEnabled:
+                ifaces[iface.Description]['up'] = True
+                for ip in iface.IPAddress:
+                    if '.' in ip:
+                        if 'inet' not in ifaces[iface.Description]:
+                            ifaces[iface.Description]['inet'] = []
+                        item = {'address': ip,
+                                'label': iface.Description}
+                        if iface.DefaultIPGateway:
+                            broadcast = next((i for i in iface.DefaultIPGateway if '.' in i), '')
+                            if broadcast:
+                                item['broadcast'] = broadcast
+                        if iface.IPSubnet:
+                            netmask = next((i for i in iface.IPSubnet if '.' in i), '')
+                            if netmask:
+                                item['netmask'] = netmask
+                        ifaces[iface.Description]['inet'].append(item)
+                    if ':' in ip:
+                        if 'inet6' not in ifaces[iface.Description]:
+                            ifaces[iface.Description]['inet6'] = []
+                        item = {'address': ip}
+                        if iface.DefaultIPGateway:
+                            broadcast = next((i for i in iface.DefaultIPGateway if ':' in i), '')
+                            if broadcast:
+                                item['broadcast'] = broadcast
+                        if iface.IPSubnet:
+                            netmask = next((i for i in iface.IPSubnet if ':' in i), '')
+                            if netmask:
+                                item['netmask'] = netmask
+                        ifaces[iface.Description]['inet6'].append(item)
+            else:
+                ifaces[iface.Description]['up'] = False
+    return ifaces
+
+
+def get_interface_info():
+    '''
+    This function will return network interface information for the system and
+    will use the best method to retrieve that information. Windows 7/2008R2 and
+    below will use WMI. Newer systems will use .NET.
+    Returns:
+        dict: A dictionary of information about the Network interfaces
+    '''
+    # On Windows 7 machines, use WMI as dotnet 4.0 is not available by default
+    if USE_WMI:
+        return get_interface_info_wmi()
+
+    # Massage the data returned by dotnet to mirror that returned by wmi
+    interfaces = get_interface_info_dot_net()
```

Comment:
you are not using the same level of abstraction here.

In one case (wmi), you return what that worker is doing. In the other (.net), you call the worker than you do more extra stuff. Maybe you should have that worker do that extra stuff? Or maybe you can have a subworker that's doing what get_interface_info_dot_net() is doing, and have get_interface_info_dot_net() call that and do this extra stuff?